crates/ruff/src/lib.rs

```diff
@@ -228,7 +228,7 @@ fn server(args: ServerCommand) -> Result<ExitStatus> {
     // by default, we set the number of worker threads to `num_cpus`, with a maximum of 4.
```

Comment:
Could you say a bit more about the problem this PR is trying to fix? According to this comment, it looks like the current way the code works is intentional?

Reply:
Unintuitively, `.max()` returns the maximum value of self and the argument (*not* limiting to the argument). To limit the value to 4, one needs to use `.min()`.

Reply:
See, e.g., https://github.com/astral-sh/ruff/blob/a2a7b1e26897e0257434b74f5734df84d7e3d5af/crates/ruff_db/src/system/os.rs#L433

Reply:
Oh I see. Right, that does make sense. Thanks!